backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IntrinsicGenerator.kt

```diff
@@ -699,10 +699,10 @@ internal class IntrinsicGenerator(private val environment: IntrinsicGeneratorEnv
     private fun FunctionGenerationContext.emitUnaryMinus(args: List<LLVMValueRef>): LLVMValueRef {
         val first = args[0]
         val destTy = first.type
-        val const0 = makeConstOfType(destTy, 0)
         return if (destTy.isFloatingPoint()) {
-            fsub(const0, first)
+            fneg(first)
```

Comment:
> Simple negation from 0 doesn't work in case -0.0 argument.
Do we have a test for that?

Reply:
Yes, I've added back "runtime_math_harmony" test that found this issue.